pyron-plugin/src/main/scala/com/cloudentity/pyron/plugin/util/value/domain.scala

```diff
@@ -26,31 +35,39 @@ object ValueOrRef {
     } getOrElse Right(Value(json))
   }
 
-  private def decodeReference(string: String): Either[String, RefType] = {
-    // FIX: check corner cases when splitting
-    string.substring(1).split('.').toList match {
-      case refType :: path =>
-        refType match {
-          case "body" =>
-            Right(BodyRef(Path(path)))
-          case "authn" =>
-            Right(AuthnRef(Path(path)))
-          case "pathParams" =>
-            Right(PathParamRef(path.mkString(".")))
-          case "headers" =>
-            Right {
-              path match {
-                case header :: "*" :: Nil =>
-                  HeaderRef(header, AllHeaderRefType)
-                case path =>
-                  HeaderRef(path.mkString("."), FirstHeaderRefType)
-              }
-            }
-          case x =>
-            Left(s"invalid reference type: $x")
-        }
-      case Nil =>
-        Left("reference cannot be empty")
+  private def decodeReference(string: String): Either[String, RefType] = string.drop(1).split('.').toList match {
+    case Nil => Left("reference cannot be empty")
+    case refType :: path => decodeRefWithPath(refType, path)
```

Comment:
Up to you, but from readability perspective consider having 2 explicit cases: ref-without-path (`case refType :: Nil`) and ref-with-path (`case refType :: path`). Personally, I have nothing against little code duplication that would be needed in this case if it simplifies the logic.

Reply:
I'll live it as it is for now. Some future PR maybe.